src/Lean/Server/InfoUtils.lean

```diff
@@ -186,33 +186,36 @@ def Info.docString? (i : Info) : MetaM (Option String) := do
 def Info.fmtHover? (ci : ContextInfo) (i : Info) : IO (Option Format) := do
   ci.runMetaM i.lctx do
     let mut fmts := #[]
-    try
-      if let some f ← fmtTerm? then
+    let decl ← try
+      let (f, decl) ← fmtTerm?
+      if let some f := f then
         fmts := fmts.push f
-    catch _ => pure ()
+      pure decl
+    catch _ => pure none
     if let some m ← i.docString? then
       fmts := fmts.push m
+    if let some decl := decl then
+      if let some f ← fmtModule? decl then
+        fmts := fmts.push f
     if fmts.isEmpty then
       return none
     else
       return f!"\n***\n".joinSep fmts.toList
 
 where
-  fmtTerm? : MetaM (Option Format) := do
+  fmtTerm? : MetaM (Option Format × Option Name) := do
```

Comment:
This looks a bit obscure to me, I would suggest repeating the `e.isConst` branch in `fmtModule?` instead

Reply:
Well it was the `whnf` that I wanted to avoid calling a second time.

Reply:
What `whnf`?

Reply:
...I must have hallucinated that. I mean `instantiateMVars`

Reply:
Ah. `docString?` seems to show that it's not really necessary for constants, but I suppose it can't hurt.